libraries/standard/http/src/http_client.c

```diff
@@ -1858,7 +1858,7 @@ static HTTPStatus_t receiveAndParseHttpResponse( const HTTPTransportInterface_t
 
 /*-----------------------------------------------------------*/
 
-HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
+HTTPStatus_t HTTPClient_Send( const TransportInterface_t * pTransport,
                               HTTPRequestHeaders_t * pRequestHeaders,
                               const uint8_t * pRequestBodyBuf,
                               size_t reqBodyBufLen,
```

Comment:
All changes above are from having a united `TransportInterface_t` for all libraries. However, I have not added the above changes to MQTT to keep this PR small.